ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -710,7 +715,14 @@ private boolean verifyJoin(
             ));
           }
 
-          return true;
+          if (!(leftExpression instanceof ColumnReferenceExp)) {
```

Comment:
We should change the error message: `Invalid join condition: foreign-key table-table joins don't support expressions in the join condition yet.` (conceptually, we can support expression for the left hand side).

However the `rightExpression` _must_ refer to the PK of the right input table and we should add a corresponding check. (I will add a check for right hand side via #7543 )